File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        hashmap={}
        temp=head
        while(temp != None):
            if temp not in hashmap:
                hashmap[temp]=1
            else:
                return True
            temp=temp.next
        return False
            
        

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode temp=head;
        if(head == null) return false;
        HashMap<ListNode,Integer> hmap = new HashMap<>();
        while(temp != null){
            if(!hmap.containsKey(temp)){
                hmap.put(temp,1);
            }
            else{
                return true;
            }
            temp=temp.next;
        }
        return false;
    }
}